shortlister/tournament.py

```diff
@@ -0,0 +1,142 @@
+from copy import deepcopy
+from itertools import combinations
+from pathlib import Path
+import pickle
+from typing import Dict, List
+
+
+def comparison(mylist, result):
+    """Starts the comparison process"""
+    pairs = get_pair(
+        mylist, result
+    )  # returns a list of pairs that can be used to compare
+    for pair in pairs:
+        print(f"{pair[0].name}[l] : {pair[1].name}[r]")
+        winner = choose(pair)
+        save_results(pair, winner, result)
+    # goes to the next pair
+
+
+def get_pair(mylist, result):
+    """Return every unique item pair in the list"""
+    unique_pairs = frozenset(
```

Comment:
Does this need to be a frozenset?

Reply:
I think only when I'm saving pair as a key to dictionary. I'm not totally sure if it still needs to be frozen, since applicant have its own hashing now.

Reply:
This puts all the combinations into a single set, it doesn't make each of the pairs a frozenset.

The point of the frozenset is to use the them as keys for dictionary lookup.

Reply:
Ah sorry, we've had that discussion and it just flew over my head. I think it's still needed to stop ```get_pair()``` from generating "B,A" pair when "A, B" already exists